File: jvm-experiments-timeperiod/src/main/java/kr/kth/timeperiod/timerange/YearCalendarTimeRange.java
package kr.kth.timeperiod.timerange;

import kr.kth.timeperiod.ITimeCalendar;
import kr.kth.timeperiod.ITimePeriod;
import lombok.extern.slf4j.Slf4j;

/**
 * kr.kth.timeperiod.timerange.YearCalendarTimeRange
 * User: sunghyouk.bae@gmail.com
 * Date: 12. 12. 23.
 */
@Slf4j
public abstract class YearCalendarTimeRange extends CalendarTimeRange {

    private static final long serialVersionUID = -4211622541447295691L;

    protected YearCalendarTimeRange(ITimePeriod period, ITimeCalendar calendar) {
        super(period, calendar);
    }

    public int getBaseMonthOfYear() {
        return getTimeCalendar().getBaseMonthOfYear();
    }

    public int getBaseYear() {
        return getStartYear();
    }
}

File: jvm-experiments-timeperiod/src/main/scala/kr/kth/timeperiod/timerange/YearCalendarTimeRange.scala
package kr.kth.timeperiod.timerange

import kr.kth.timeperiod._

/**
 * kr.kth.timeperiod.timerange.YearCalendarTimeRange
 * User: sunghyouk.bae@gmail.com
 * Date: 13. 1. 16.
 */
abstract class YearCalendarTimeRange(period: ITimePeriod, calendar: ITimeCalendar)
	extends CalendarTimeRange(period, calendar) {

	def getYearBaseMonth = getTimeCalendar.getBaseMonthOfYear

	def getBaseYear = getStartYear
}
